src/cursor/cur_hs.c

```diff
@@ -142,6 +145,52 @@ __wt_hs_cursor_search_near(WT_SESSION_IMPL *session, WT_CURSOR *cursor, int *exa
     return (ret);
 }
 
+/*
+ * __curhs_next --
+ *     WT_CURSOR->next method for the hs cursor type.
+ */
+static int
+__curhs_next(WT_CURSOR *cursor)
+{
+    WT_CURSOR *file_cursor;
+    WT_CURSOR_HS *hs_cursor;
+    WT_DECL_RET;
+    WT_SESSION_IMPL *session;
+
+    hs_cursor = (WT_CURSOR_HS *)cursor;
+    file_cursor = hs_cursor->file_cursor;
+    CURSOR_API_CALL(cursor, session, next, CUR2BT(file_cursor));
```

Comment:
I think I'm using CURSOR_API_CALL_PREPARE_ALLOWED. This allows the session that runs prepared operation also to call this.

Reply:
Fixed.